### ISSUE TITLE
tf.keras computes incorrect loss values with Masking

### ISSUE THREAD

**Issue by qlzh727** [keras-team/tf-keras#130]
**This issue is copied from https://github.com/tensorflow/tensorflow/issues/34491. **

The issue is moved here for better tracking since the keras code has been moved to keras-team/keras repo.

**Comment by tilakrayal** [comment]
I tried to execute the code with the latest keras version and observed that the binary_crossentropy both as loss and metric are almost similar. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/8f4b6607e2a1e2562f08d93a056e96b9/untitled2444.ipynb). Thank you!
